In [26]:
import pandas as pd


In [22]:
titanic = pd.read_csv('/Users/lisatosti/Documents/GitHub/coding-temple/week 5/assignments/titanic.csv')
titanic.head(10)

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses_Aboard  Parents/Children_Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [23]:
titanic.columns = titanic.columns.str.lower().str.replace(' ', '_').str.replace('/', '_')
titanic.head(10)

survived  pclass                                               name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      sex   age  siblings_spouses_aboard  parents_children_aboard     fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [24]:
connection = 'postgresql://tgxxufqc:pnyU-FMki7VmrpFvAe9YtmbCt-OlBGU1@suleiman.db.elephantsql.com/tgxxufqc'


In [25]:
titanic.to_sql('titanic', con = connection, if_exists='fail')

887

### Question 1: Count how many rows you have

```sql
SELECT COUNT(*)
FROM titanic;
```
887

### Question 2: How many people survived?

```sql
SELECT COUNT(survived)
FROM titanic
WHERE survived = 1;
```
342

You can also do a SUM instead of a count but they both are the same answer.

### Question 3: What passendger class has the largest population?

```sql
SELECT pclass, COUNT(pclass) AS pclass_count
FROM titanic
GROUP BY pclass
ORDER BY pclass
DESC
LIMIT 1;
```
pclass 3 has the most passengers with 487.


### Bonus: Give me all the passengers that had their names start with Miss

```sql
SELECT *
FROM titanic
WHERE name LIKE 'Miss%';

```